### PR TITLE
team-level std algos: part 13

### PR DESCRIPTION
**Pull request by fnrizzi**
Co-authored-by: Cezary Skrzyński
Co-authored-by: Jakub Strzebonski

## Content 

Team-level implementation and tests for:

- `Kokkos_InclusiveScan.hpp`
- `Kokkos_TransformInclusiveScan.hpp`

## Technical comments

- the change `class` -> `typename` and the enable if defaulted in the template parameters comes from [this request](https://github.com/kokkos/kokkos/pull/6200#issuecomment-1626406816)

- the code duplication is intentional and agreed upon after the long discussion in [here](https://github.com/kokkos/kokkos/pull/6200)

- tests are run for both contiguous views and strided views

## Other comments

- all team-level PRs are tracked in: https://github.com/kokkos/kokkos/issues/6352

- this comes from breaking apart https://github.com/kokkos/kokkos/pull/5150

## Possible conflicts

Implementation-wise, this has not conflict with other team-level PRs so it could be merged independently. 
However, a trivial conflict in the CMakeLists inside the unit test occurs if another of the team-level PRs is merged before this because each team-level PR adds test using a specific "label". Resolving this is trivial. 

## Associated wiki PRs

- `Kokkos_InclusiveScan.hpp`, `Kokkos_TransformInclusiveScan.hpp`: see [wiki PR](https://github.com/kokkos/kokkos-core-wiki/pull/440)

IMPORTANT: wiki PRs are not complete yet but 99% there 

The issue tracking all *wiki PRs* is [here](https://github.com/kokkos/kokkos-core-wiki/issues/379)

### CODE REVIEW

**Comment by fnrizzi** [comment]
Retest this please

File: algorithms/unit_tests/CMakeLists.txt [hunk 1]
```diff
@@ -153,6 +153,18 @@ foreach(Tag Threads;Serial;OpenMP;Cuda;HPX;HIP;SYCL;OpenMPTarget)
...
 
+    # ------------------------------------------
+    # std team Q
+    # ------------------------------------------
+    set(STDALGO_TEAM_SOURCES_Q)
+    foreach(Name
+  StdAlgorithmsCommon
+  StdAlgorithmsTeamInclusiveScan
+  StdAlgorithmsTeamTransformInclusiveScan
+  )
```
Comment (masterleinad) [inline algorithms/unit_tests/CMakeLists.txt:164]: Again, weird indentation. You used tabs before.

File: algorithms/unit_tests/TestStdAlgorithmsTeamInclusiveScan.cpp [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    auto srcRow    = Kokkos::subview(sourceViewBeforeOp_h, i, Kokkos::ALL());
+    auto first     = KE::begin(srcRow);
+    auto last      = KE::end(srcRow);
+    auto destRow   = Kokkos::subview(stdDestView, i, Kokkos::ALL());
+    auto firstDest = KE::begin(destRow);
+    auto initValue = initValuesView_h(i);
+
+    ASSERT_TRUE(intraTeamSentinelView_h(i));
+
+#if defined(__GNUC__) && __GNUC__ == 8
```
Comment (masterleinad) [inline algorithms/unit_tests/TestStdAlgorithmsTeamInclusiveScan.cpp:214]: Again, check for the standard library version, not the compiler.

Comment (fnrizzi) [reply algorithms/unit_tests/TestStdAlgorithmsTeamInclusiveScan.cpp:214]: Yes, leftover from before , will fix it thank you for pointing it out!

File: algorithms/unit_tests/CMakeLists.txt (hunk 1 above)
Comment (fnrizzi) [reply algorithms/unit_tests/CMakeLists.txt]: fixed

File: algorithms/unit_tests/TestStdAlgorithmsTeamInclusiveScan.cpp (hunk 2 above)
Comment (fnrizzi) [reply algorithms/unit_tests/TestStdAlgorithmsTeamInclusiveScan.cpp]: fixed

**Review by masterleinad — APPROVED** [review]

**Review by nliber — APPROVED** [review]

**Pull request merged**